EIPS/eip-5792.md

```diff
@@ -207,23 +169,66 @@ As with the return value of `wallet_sendCalls`, the batch identifier may be any
 }
 ```
 
-### `atomicBatch` Capability
+### `wallet_getCapabilities`
+
+An app calling the above-defined methods without prior knowledge that the wallet supports them risks degrading user experience considerably, so an additional RPC for querying the wallet's capabilities is provided.
 
-Indicates that a wallet can execute multiple calls atomically as part of a single transaction.
+In addition to, or instead of, querying the wallet directly for capabilities, the same capability objects could be expressed out-of-band (such as in the `sessionProperty.capabilities` object persisted in a CAIP-25-conformant wallet provider interface) or in a well-known location (such as a URL derived from an [EIP-6963](./eip-6963.md) `rdns` identifier). Provider abstractions may also cache capabilities from previous requests or otherwise inject them from out-of-band. If querying multiple ways leads to different results, or to override caches or provider injections, direct RPC communication with the wallet SHOULD always be taken as the canonical and current expression of capabilities.
```

Comment:
Does CAIP-25 have a field for this? Or expressly allow new sets of fields defined by the wallet?

Reply:
e.g. would this conform to the spec? 
> Respondent SHOULD ignore and drop from its response any properties not defined in this document or in another CAIP document extending this protocol which the respondent has implemented in its entirety; similarly, the requiredScopes, optionalScopes, and sessionScopes arrays returned by the respondent SHOULD contain only valid [CAIP-217](https://chainagnostic.org/CAIPs/caip-217) objects, and properties not defined in [CAIP-217](https://chainagnostic.org/CAIPs/caip-217) SHOULD also be dropped from each of those objects.

Reply:
> Does CAIP-25 have a field for this? Or expressly allow new sets of fields defined by the wallet?

there is, in fact, a property for exactly this:

> A third object is the sessionProperties object, all of whose properties MUST be interpreted as optional, since requesting applications cannot mandate session variables to providers. Because they are optional, providers MAY respond with all of the requested properties, or a subset of the session properties, or no sessionProperties object at all; they MAY even replace the values of the optional session properties with their own values.

The semantics and syntax of *specific* members of the `sessionProperties`, such as `sessionProperties.capabilities`, are specified by future CAIPs, in the same way that the semantics and syntax additional capability object are expected to be specified in future EIPs.  Ideally, 1:1 equivalence and translatability between the two should be preserved, which is why there aren't any CAIPs defining these yet 😉

Reply:
(in the meantime, experimental implementations are breaking one SHOULD, but no MUSTs)

Reply:
Ok, but I think this leaves us in an odd state: we are saying 
> such as in the `sessionProperty.capabilities` object persisted in a CAIP-25-conformant wallet
and then also that 
> The semantics and syntax of specific members of the sessionProperties object, such as sessionProperties.capabilities, are expected to be specified by future CAIPs

So isn't it the case that, today, one could not properly follow CAIP-25 and include `sessionProperty.capabilities` ?

Reply:
well i was waiting for this EIP to be stable to push that CAIP to draft, but you could CONFORMANTLY follow CAIP-25 as either a wallet or a dapp putting anything you want in `sessionProperties`. I wouldn't advice people do it in prod until there's a CAIP they trust their counterparties to also implement, that's kind of the proleptic temporality of EIP adoption that led us to such proactive providers in the first place, isn't it 😏

Reply:
Roger. Would it make sense to get that CAIP-25 update in and then we can include it more explicitly here as an alternative means of discovery?

Reply:
well, if by "more explicitly" you mean normatively refer, I don't think it's a good idea, since we're explicitly telling people it's supplemental and _not an alternative or a replacement_ to the wallet_RPC, so the reference should only be NON-normative (and thus linking or not linking doesn't matter).  it's already happening in the world, and you have to name to mark it as supplemental (especially if you'd like to deprecate it later).

i also don't love the idea of removing the reference to it (or the reference to provider-caching, more importantly, which I think wallet devs and dapp devs alike should be given a heads-up about, honestly).

Reply:
(and technically, it wouldn't be a CAIP-25 update, it was be a novel _extension_ CAIP that defines sessionProperties.capabilities. I suppose I could tho, if this EIP isn't going to specify anything more concrete, syntactically or semantically, than `The key of each item per chain is the name of a capability and the value can be of any shape. Capabilities are returned per chain because wallets may support different capabilities across chains.`.  It'll be the shortest CAIP yet, until future EIPs add more sub-classes!)